spring-batch-core/src/main/java/org/springframework/batch/core/configuration/support/DefaultBatchConfiguration.java

```diff
@@ -187,9 +192,17 @@ public JobExplorer jobExplorer() throws BatchConfigurationException {
 		}
 	}
 
-	@Bean
+	/**
+	 * Defines a {@link JobRegistry} bean.
+	 * @return a {@link JobRegistry} bean
+	 * @throws BatchConfigurationException if unable to register the bean
+	 * @since 5.0
+	 * @deprecated in favor of bean injection, and declaring a custom bean with name
+	 * {@code jobRegistry}
```

Comment:
I am trying to test this from a user's point of view (which is also what is mentioned in the [docs](https://docs.spring.io/spring-batch/reference/job/advanced-meta-data.html#jobregistry)) against this PR but it fails (while it passes on the current `main`). Here is a patch that slightly modifies the canonical example I used for tests: 

[test_pr4520.patch](https://github.com/spring-projects/spring-batch/files/13991872/test_pr4520.patch)